scripts/genVis.py

```diff
@@ -1,58 +1,140 @@
+__author__ = 'Jon Mallen'
+
 from sklearn.manifold import TSNE
-#from MulticoreTSNE import MulticoreTSNE as TSNE
 from sklearn.decomposition import PCA
-import numpy as np
+import umap
 import matplotlib.pyplot as plt
-import matplotlib.cm as cm
 import pandas as pd
-from optparse import OptionParser 
-parser = OptionParser()
+import argparse
+import itertools
 
-parser.add_option('-i','--inFile',default='',type=str,
-                  help='Specify input expression matrix file name')
+# Define arguments
+parser = argparse.ArgumentParser("Visualize the BoolODE simulation of cells.")
+parser.add_argument('-f', '--pathToFiles', default='', type=str, help='Specify path to files.')
+parser.add_argument('-p', '--pca', nargs='*', help='Call to visualize with PCA. Specify dimension (2 or 3) as argument.'
+                                                   'Default is 2.')
+parser.add_argument('-t', '--tsne', nargs='*', help='Call to visualize with t-SNE. '
+                                                    'Specify dimension (2 or 3) as argument. Default is 2.')
+parser.add_argument('-u', '--umap', nargs='*', help='Call to visualize with UMAP. '
+                                                    'Specify dimension (2 or 3) as argument. Default is 2.')
+parser.add_argument('-c', '--handleExternalClusterFile', action='store_true', default=False,
+                    help='Call to handle an external cluster file.')
+parser.add_argument('-n', '--plotTitle', default='', nargs='*', help='Write a plot title.')
```

Comment:
If I make this to be `-t` (for title), do you have any preference for what I would change the argument for t-SNE to be, since the t-SNE argument already uses `-t`? Alternatively, I can change my description to be "Name of the plot" so that it is `-n` (for name).